### PR TITLE
Added ArrayBuffer and Blob as data types

### LINKED ISSUE #28: Not able to get binary messages
Binary messages does have the length but actual blob does not have the message binary blob. It seems like the message is always observable with type of message is string only (index.ts, line 8) = "messages: Observable&lt;string&gt;"

### PR DESCRIPTION
**Pull request by saitonakamura**
Added multiple variants of incoming and outcoming data, instead of just `string` and gave a way to narrow to down via generic. It's not super accurate as #49 tried to achieve, but it's something one can work with.

It's a breaking change unfortunately, but I can't think of any sane way to mitigate it (defaulting generic to `string` doens't look good to me). Let me know if you're comfortable with such changes, so I can update README with the new info.

Fixes #28

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import { Observable, Subscription, BehaviorSubject } from 'rxjs'
 
-export interface Connection {
+export interface Connection<T extends string | ArrayBuffer | Blob = string | ArrayBuffer | Blob> {
```
Comment (ohjames) [inline src/index.ts]: `ArrayBuffer` appears twice and also not sure why both `Blob` and `Blob = string` are specified. Not sure what the latter means.

File: src/index.ts [hunk 2]
```diff
@@ -20,15 +20,15 @@ const defaultProtocols = [];
 
 const defaultWebsocketFactory: WebSocketFactory = (url: string, protocols: string | string[] = defaultProtocols): IWebSocket => new WebSocket(url, protocols)
 
-export default function connect(
+export default function connect<T extends string | ArrayBuffer | Blob = string | ArrayBuffer | Blob>(
```
Comment (ohjames) [inline src/index.ts]: Same comment as before.

File: src/index.ts (hunk 1 above)
Comment (saitonakamura) [reply src/index.ts]: Operators precedence with a bit different here, `=` takes higher place. Essentially it's generic argument `T` with a constraint of `string | ArrayBuffer | Blob` **and** default value of `string | ArrayBuffer | Blob`. Without the latter you'll have to specify generic argument every time.

Comment (ohjames) [reply src/index.ts]: Doah of course, don't suppose it's possible to add `()` around the union types to make that more clear?

Comment (ohjames) [reply src/index.ts]: And yeah I get that `=` has lower precedence (rather than higher) now, I should have really realised that in the first place ;)

Comment (saitonakamura) [reply src/index.ts]: > don't suppose it's possible to add () around the union types to make that more clear?

I didn't know that, but it is in fact possible! Added parenthesis.

> And yeah I get that = has lower precedence

Yeah, you're right, sorry for messing it up even more :)

**Comment by ohjames** [comment]
Cool I need to make sure circleci is running on the PRs then if the validations pass this should be good to merge.

**Comment by ohjames** [comment]
Got the circleci checks running!

**Comment by ohjames** [comment]
I've squashed your commits and added them to the v7 branch: https://github.com/ohjames/rxjs-websockets/tree/v7

I'm gonna improve/simplify the connection status handling here too, better to add two backwards incompatible changes in a single release.

**Comment by ohjames** [comment]
I've released `7.0.0-alpha.1` with your change. There will be one more backwards incompatible change before `7.0.0` is released.

**Comment by ohjames** [comment]
I'm closing this as your code is in the latest alpha and properly attributed to you. The official 7 release will be out shortly and also includes much simpler and more powerful handling of connection status observation.

**Comment by saitonakamura** [comment]
Cool, i'll be waiting

**Comment by ohjames** [comment]
The API is stable now with the release of `7.0.0-beta.1`. 

See https://github.com/ohjames/rxjs-websockets/tree/v7 for the documentation. Would be great to hear your feedback! I think this new API is much better :new_moon_with_face: :cactus: :cat: :cat2:

**Pull request closed without merging**